Add documentation for user management (#3320)
treeverse/lakeFS: docs/reference/authentication.md

`AKIA...` and an associated secret access key.  These credentials are valid
to log into the Web UI, or to authenticate programmatic requests to the API
Server or the S3 Gateway.

> git

Reviewer: I don't understand this line.

Author: My mistake. (Probably just tried to type `git` in the command line and accidentally typed it into the doc). Fixed

> #### lakeFS Invite

Reviewer: Nit: Please add a blank line above.

Author: Done

> #### lakeFS Invite

Reviewer: Something is a bit off with this title.  Perhaps

```suggestion
#### Invited to lakeFS?
```

@adipolak - can you help?

Author: Not sure I agree on this one. I think @adipolak should decide on this one

> If you were invited to lakeFS, Follow these steps to create and activate your account:

Reviewer: As mentioned in this review, please correct spelling and capitalization.

Author: Done.

> If you were invited to lakeFS, Follow these steps to create and activate your account:

Reviewer: We should _first_ explain that the invitation will be sent by email.

Author: Done.

> 1. Locate the email sent to you, (be aware it may be in your spam folder)

Reviewer: Please read and follow our [documentation guidelines](https://docs.google.com/document/d/1vif05nDiKZYPwYdINLoq4BJ6j7bBIQ6ZA2dt7McuzyY/edit#heading=h.vuqinl3vgds3) about "How to create a list".  Here punctuation is missing.

Author: Done

> 1. Click on the link in the email, to be directed to the account activation page

Reviewer: The second half of the sentence is in the passive voice.  See [style basics](https://docs.google.com/document/d/1vif05nDiKZYPwYdINLoq4BJ6j7bBIQ6ZA2dt7McuzyY/edit#heading=h.dqp77semtfge).

```suggestion
1. Click on the link in the email to open the account activation page.
```

Author: Both are active voices in my opinion, but I do accept the change

1. Set your account password
1. Now you can login to the UI using your email and the password you created for your account in the previous step

> Note that email/password credentials should only be used for login via the UI.

Reviewer: Please rephrase in a more positive mode: IIUC what this is saying is that you can only use email and password to login to the GUI.  So you should also say how to connect to everything else.

Also, I am not sure that this is correct: can I use email and password in `lakectl`?  Because that is also our UI.  If I *cannot* use email and password in `lakectl`, then this should be "GUI".

Author: Added the point that a set of access and secret keys should be generated.
Using the email/password is not supported for anything besides the GUI. (It will technically work though). Fixed to GUI.

to log into the Web UI, or to authenticate programmatic requests to the API
Server or the S3 Gateway.

> #### LakeFS Invite

Reviewer: Please revert this change, per @adipolak 's already-mentioned [https://docs.google.com/document/d/1vif05nDiKZYPwYdINLoq4BJ6j7bBIQ6ZA2dt7McuzyY/](guide).  The spelling "lakeFS" is explicitly mentioned there:

> Make sure to spell the brand name correctly: **lakeFS**. Not LakeFS or lakefs.

The first word on the website lakefs.io is "lakeFS".  If you go to @peacing 's [latest blog](https://lakefs.io/towards-effective-dataops/), you can find "lakeFS" at the start of a sentence, _in a heading_.

The inconsistent capitalization in the heading is on the _second_ word.  This document is consistent in using a single capital at the start of every heading at level 4.  And AFAICS this is a consistent practice on our documentation.  The correct heading is therefore

```suggestion
#### lakeFS invite
```

**BUT:** Please just leave it alone and pull this PR.  We can fix it later.

Thanks!

to log into the Web UI, or to authenticate programmatic requests to the API
Server or the S3 Gateway.

The built-in authentication database supports authentication to the UI, via
email and password as well. This option is only valid for users who have 
been invited to lakefs, confirmed the invite, and created a password.
These credentials should be used for UI access only. It should be noted, that 
the initial admin user who created the repo, will not be able to use email/password 

> authentiaction.

Reviewer: ```suggestion
authentication.
```
Please spell-check documentation.

> 

Reviewer 2: 1. For this to work, you need to enable an email server integration, right?
2. In general, note that this feature is interesting to both the system admin (the one who installs lakeFS), and to the end user. You should be more mindful of this.
3. "the admin user who created the repo" >> "the initial lakeFS admin user" - the user didn't necessarily create a repo.
4. Also see the [documentation guidelines](https://docs.google.com/document/d/1vif05nDiKZYPwYdINLoq4BJ6j7bBIQ6ZA2dt7McuzyY/edit): it has some helpful suggestions on which language to use. In particular, the branch name should be spelled as _lakeFS_.

Reviewer 3: +1 for 1.
I think it's important to mention that using the email/password authentication requires additional configuration (and either provide a link to this configuration or explain it inline, IMO link to the full documentation is preferred).
